sansa-owl-spark/src/main/scala/net/sansa_stack/owl/spark/stats/OWLStats.scala

```diff
@@ -0,0 +1,821 @@
+      .map(a => ((a.getSubject.asOWLNamedIndividual().getIRI.getNamespace, a.getObject.asOWLNamedIndividual().getIRI.getNamespace), 1))
+
+    // data property assertion axioms of named individuals
+    val dataAssr = owlAxioms.extractAxioms(axioms, AxiomType.DATA_PROPERTY_ASSERTION)
+      .asInstanceOf[RDD[OWLDataPropertyAssertionAxiom]]
+      .filter(_.getSubject.isNamed)  // to be able to safely call .asOWLNamedIndividual
+//      .filter(a => a.getSubject.asOWLNamedIndividual().getIRI.getNamespace != a.getObject.getDatatype.getIRI.getNamespace)  // this check is not needed as the set of individuals and the set of datatypes are disjoint
+      .map(a => ((a.getSubject.asOWLNamedIndividual().getIRI.getNamespace, a.getObject.getDatatype.getIRI.getNamespace), 1))
+
+    val links = spark.sparkContext.union(
+        declarationAxiomLinks, subClassAxiomLinks, disjointClassesAxiomLinks, disjointObjPropAxiomLinks,
+        disjointDataPropAxiomLinks, subObjPropAxiomLinks, subDataPropAxiomLinks, subAnnPropAxiomLinks,
+        objPropDomainAxiomLinks, dataPropDomainAxiomLinks, annPropDomainAxiomLinks, objPropRangeAxiomLinks,
+        dataPropRangeAxiomLinks, annPropRangeAxiomLinks, eqObjectPropAxiomLinks, eqDataPropAxiomLinks,
+        invObjPropAxiomLinks, sameIndvAxiomLinks, diffIndvAxiomLinks, classAssrAxiomLinks, objAssrAxiomLinks, dataAssr)
+      .reduceByKey(_ + _)
+      .map(a => (a._1._1, a._1._2, a._2))
+
+    links
+  }
+
+  /**
+    * 28.Maximum value per property {int,float,double} criterion
+    *
+    * @param axioms RDD of OWLAxioms
+    * @return entities with their maximum values
+    */
+  def getMaxPerNumericDatatypeProperty(axioms: RDD[OWLAxiom]): RDD[(OWLDataPropertyExpression, Int)] = {
+
+    val dataPropAssrNumbers = owlAxioms.extractAxioms(axioms, AxiomType.DATA_PROPERTY_ASSERTION)
+                                .asInstanceOf[RDD[OWLDataPropertyAssertionAxiom]]
+                                .filter(a =>
+                                  a.getObject.getDatatype.isInteger ||
+                                    a.getObject.getDatatype.isDouble ||
+                                    a.getObject.getDatatype.isFloat)
+                                .map(a => (a.getProperty, a.getObject.getLiteral.toInt))
+
+    val negDataPropAssrNumbers = owlAxioms.extractAxioms(axioms, AxiomType.NEGATIVE_DATA_PROPERTY_ASSERTION)
+                                   .asInstanceOf[RDD[OWLNegativeDataPropertyAssertionAxiom]]
+                                   .filter(a =>
+                                      a.getObject.getDatatype.isInteger ||
+                                        a.getObject.getDatatype.isDouble ||
+                                        a.getObject.getDatatype.isFloat)
+                                   .map(a => (a.getProperty, a.getObject.getLiteral.toInt))
+
+    val max = dataPropAssrNumbers.union(negDataPropAssrNumbers)
```

Comment:
Does it make sense to generate a union of data property assertions and negative data property assertions? If an ontology describes persons and their heights, it would be totally fine to add the negative data property assertion that personX is not 42 meters high. However, you would still get the 42 meters as the maximum then.

Reply:
Same holds for the average calculation below.

Reply:
So the right way is only to consider the `DataPropertyAssertions`?